README.md
# Paleta

[![License Apache 2][badge-license]](LICENSE)
[![GitHub version](https://badge.fury.io/gh/pilotariak%2Fpaleta.svg)](https://badge.fury.io/gh/pilotariak%2Fpaleta)

* Master : [![Circle CI](https://circleci.com/gh/pilotariak/paleta/tree/master.svg?style=svg)](https://circleci.com/gh/pilotariak/paleta/tree/master)
* Develop : [![Circle CI](https://circleci.com/gh/pilotariak/paleta/tree/develop.svg?style=svg)](https://circleci.com/gh/pilotariak/paleta/tree/develop)

This tool is a simple CLI to display informations about Pelota competitions for the terminal.
Supported leagues are :

* [x] [Côte d'Argent](http://www.lcapb.net/)
* [x] [Pays Basque](http://www.comite-pelote-basque.eus/fr/)
* [ ] [Landes](http://www.llpb.fr/)
* [x] [Béarn](http://liguebearnpelote.fr/)
* [ ] [Midi-Pyrénées](http://www.lmppb.fr/)
* [x] [Ile de France](http://www.lidfpb.fr/)

* [x] [Fédération française](http://www.ffpb.net/)


![Screenshot](paleta.png)


## Installation

You can download the binaries :

* Architecture i386 [ [linux](https://bintray.com/artifact/download/pilotariak/oss/paleta-0.5.0_linux_386) / [darwin](https://bintray.com/artifact/download/pilotariak/oss/paleta-0.5.0_darwin_386) / [freebsd](https://bintray.com/artifact/download/pilotariak/oss/paleta-0.5.0_freebsd_386) / [netbsd](https://bintray.com/artifact/download/pilotariak/oss/paleta-0.5.0_netbsd_386) / [openbsd](https://bintray.com/artifact/download/pilotariak/oss/paleta-0.5.0_openbsd_386) / [windows](https://bintray.com/artifact/download/pilotariak/oss/paleta-0.5.0_windows_386.exe) ]
* Architecture amd64 [ [linux](https://bintray.com/artifact/download/pilotariak/oss/paleta-0.5.0_linux_amd64) / [darwin](https://bintray.com/artifact/download/pilotariak/oss/paleta-0.5.0_darwin_amd64) / [freebsd](https://bintray.com/artifact/download/pilotariak/oss/paleta-0.5.0_freebsd_amd64) / [netbsd](https://bintray.com/artifact/download/pilotariak/oss/paleta-0.5.0_netbsd_amd64) / [openbsd](https://bintray.com/artifact/download/pilotariak/oss/paleta-0.5.0_openbsd_amd64) / [windows](https://bintray.com/artifact/download/pilotariak/oss/paleta-0.5.0_windows_amd64.exe) ]
* Architecture arm [ [linux](https://bintray.com/artifact/download/pilotariak/oss/paleta-0.5.0_linux_arm) / [freebsd](https://bintray.com/artifact/download/pilotariak/oss/paleta-0.5.0_freebsd_arm) / [netbsd](https://bintray.com/artifact/download/pilotariak/oss/paleta-0.5.0_netbsd_arm) ]


## Usage

* CLI help:

        $ paleta help
        NAME:
        paleta - CLI for Pelota news

        USAGE:
            paleta [global options] command [command options] [arguments...]

        VERSION:
            0.5.0

        COMMANDS:
            version
            leagues
            league
            help, h  Shows a list of commands or help for one command

        GLOBAL OPTIONS:
            --debug        Enable debug mode
            --help, -h     show help
            --version, -v  print the version

* Show supported leagues :

        $ paleta leagues list
        Leagues:
        - ctpb
        - lbpb
        - lcapb
        - lidfpb

* Display informations about a leage:

        $ paleta league describe --league lcapb
        +--------------+-----------------------------------------+
        | Name         | LIGUE DE PELOTE BASQUE DE CÔTE D’ARGENT |
        +--------------+-----------------------------------------+
        | Address      | Maison Départementale des Sports        |
        |              | 153, rue David Johnston                 |
        |              | 33000 Bordeaux                          |
        +--------------+-----------------------------------------+
        | Website      | http://www.lcapb.net/                   |
        +--------------+-----------------------------------------+
        | Email        | contact@lcapb.net                       |
        +--------------+-----------------------------------------+
        | Phone number | 05 56 00 99 15                          |
        +--------------+-----------------------------------------+

* Show available challenge:

        $ paleta league challenges --league lcapb
        Challenges:
        - [20180501] Championnat 2017-2018 CCAPB
        - [20170501] Championnat 2016-2017 CCAPB
        - [20160501] Championnat Hiver 2015-2016 LCAPB
        - [20150501] Championnat Hiver 2014-2015 LCAPB
        - [20130501] Championnat Hiver 2013-2014 LCAPB
        - [20190501] Championnat 2018-2019 CCAPB

* Show available levels:

        $ paleta league levels --league lcapb
        Levels:
        - [51] Senoir Individuel
        - [1] 1ère Série
        - [2] 2ème Série
        - [3] 3ème Série
        - [4] Seniors
        - [6] Cadets
        - [9] Poussins
        - [7] Minimes
        - [8] Benjamins

* Show available disciplines:

        $ paleta league disciplines --league lcapb
        Disciplines:
        - [2] Trinquet / P.G. Pleine Masculin
        - [4] Trinquet / P.G. Pleine Feminine
        - [26] Mur à Gauche / P.G. Pleine Masculin
        - [126] Mur A gauche / P.G. Pleine Masculin Barrages
        - [501] Place Libre / P.G Pleine Feminine
        - [3] Trinquet / P.G. Creuse Masculin
        - [5] Trinquet / P.G. Creuse Feminine
        - [13] Place Libre / Grand Chistera
        - [16] Place Libre / P.G. Pleine Masculin
        - [27] Mur à Gauche / P.G. Pleine Feminine
        - [28] Mur à Gauche / P.G. Creuse Masculin Individuel

* Display result for a competiion:

        $ paleta league results --league lcapb --level 2 --discipline 2 --challenge 20190501
        +---------------+--------------------------------------+--------------------------------------+-------+-------------+
        |     DATE      |                CLUB 1                |                CLUB 2                | SCORE | COMMENTAIRE |
        +---------------+--------------------------------------+--------------------------------------+-------+-------------+
        | Poules 1      | Club 1                               | Club 2                               | Score | Commentaire |
        +---------------+--------------------------------------+--------------------------------------+-------+-------------+
        | 02/09/2018    | C.A. BEGLAIS                         | S.A. GAZINET CESTAS                  | 40/24 |             |
        |               | (073842) GARCIA Antoine (S)          | (080163) LESTE Maxime                |       |             |
        |               | (064403) CHARPENTIER Mathieu         | (030552) MICHALUC Christophe         |       |             |
        +---------------+--------------------------------------+--------------------------------------+-------+-------------+
        | 09/09/2018    | S.A. GAZINET CESTAS                  | PILOTARI IRRATZABAL CLUB             | 40/15 |             |
        |               | (080163) LESTE Maxime                | (093680) MENVIELLE Bertrand          |       |             |
        |               | (030552) MICHALUC Christophe         | (093679) ROBIN-BROSSE David          |       |             |
        +---------------+--------------------------------------+--------------------------------------+-------+-------------+
        ...
        +---------------+--------------------------------------+--------------------------------------+-------+-------------+
        | 1/2 finale    | Club 1                               | Club 2                               | Score | Commentaire |
        +---------------+--------------------------------------+--------------------------------------+-------+-------------+
        | 28/10/2018    | BORDEAUX ETUDIANTS CLUB              | PILOTARI CLUB VILLENAVAIS            | 40/39 |             |
        |               | (059429) GILLERON Pascal             | (064884) NICOL Thomas                |       |             |
        |               | (039813) LAPEYRE Clément             | (091825) THOBERT Jerome              |       |             |
        +---------------+--------------------------------------+--------------------------------------+-------+-------------+
        | 28/10/2018    | C.A. BEGLAIS                         | BORDEAUX ETUDIANTS CLUB              | 27/40 |             |
        |               | (073842) GARCIA Antoine (S)          | (088281) BOISSIERE Guillaume         |       |             |
        |               | (064403) CHARPENTIER Mathieu         | (097865) OYHENART David              |       |             |
        +---------------+--------------------------------------+--------------------------------------+-------+-------------+
        | Finale        | Club 1                               | Club 2                               | Score | Commentaire |
        +---------------+--------------------------------------+--------------------------------------+-------+-------------+
        | 04/11/2018    | BORDEAUX ETUDIANTS CLUB              | BORDEAUX ETUDIANTS CLUB              | 37/40 |             |
        |               | (059429) GILLERON Pascal             | (088281) BOISSIERE Guillaume         |       |             |
        |               | (039813) LAPEYRE Clément             | (097865) OYHENART David              |       |             |
        +---------------+--------------------------------------+--------------------------------------+-------+-------------+



## Development

* Initialize environment

        $ make init

* Build tool :

        $ make build

* Launch unit tests :

        $ make test

## Contributing

See [CONTRIBUTING](CONTRIBUTING.md).


## License

See [LICENSE](LICENSE) for the complete license.


## Changelog

A [changelog](ChangeLog.md) is available


## Contact

Nicolas Lamirault <nicolas.lamirault@gmail.com>

[badge-license]: https://img.shields.io/badge/license-Apache2-green.svg?style=flat


ChangeLog.md
ChangeLog
==============

# Version 0.5.0 (01/11/2019)

- Challenges: update to 2018/2019
- Logging: replace logging library
- Golang: refactoring using Golang modules

# Version 0.4.0 (09/20/2017)

- Add challenge 2017-2018
- Replace *govendor* by *dep*

# Version 0.3.0 (12/14/2016)

- Refactoring CLI with subcommands
- Setup continuous integration using CircleCI

# Version 0.2.0 (11/28/2016)

- Add feature to display informations about a league
- Add output colors in table
- Add LBPB league
- Add LIDFPB league
- Add CTPB league

# Version 0.1.0 (11/28/2016)

- Add LCAPB league
- Init CLI


leagues/lcapb/lcapb.go
// Copyright (C) 2016, 2017 Nicolas Lamirault <nicolas.lamirault@gmail.com>

// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at

//     http://www.apache.org/licenses/LICENSE-2.0

// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

package lcapb

import (
	// "fmt"
	// "os"

	"github.com/Sirupsen/logrus"
	// "github.com/olekukonko/tablewriter"

	"github.com/pilotariak/paleta/leagues"
)

const (
	uri = "http://lcapb.euskalpilota.fr/resultats.php"
)

var (
	challenges = map[string]string{
		"20190501": "Championnat 2018-2019 CCAPB",
		"20180501": "Championnat 2017-2018 CCAPB",
		"20170501": "Championnat 2016-2017 CCAPB",
		"20160501": "Championnat Hiver 2015-2016 LCAPB",
		"20150501": "Championnat Hiver 2014-2015 LCAPB",
		"20130501": "Championnat Hiver 2013-2014 LCAPB",
	}

	disciplines = map[string]string{
		"2":   "Trinquet / P.G. Pleine Masculin",
		"3":   "Trinquet / P.G. Creuse Masculin",
		"4":   "Trinquet / P.G. Pleine Feminine",
		"5":   "Trinquet / P.G. Creuse Feminine",
		"13":  "Place Libre / Grand Chistera",
		"16":  "Place Libre / P.G. Pleine Masculin",
		"26":  "Mur à Gauche / P.G. Pleine Masculin",
		"27":  "Mur à Gauche / P.G. Pleine Feminine",
		"28":  "Mur à Gauche / P.G. Creuse Masculin Individuel",
		"126": "Mur A gauche / P.G. Pleine Masculin Barrages",
		"501": "Place Libre / P.G Pleine Feminine",
	}

	levels = map[string]string{
		"1":  "1ère Série",
		"2":  "2ème Série",
		"3":  "3ème Série",
		"4":  "Seniors",
		"6":  "Cadets",
		"7":  "Minimes",
		"8":  "Benjamins",
		"9":  "Poussins",
		"51": "Senoir Individuel",
	}
)

func init() {
	leagues.RegisterLeague("lcapb", newLCAPBLeague)
}

type lcapbLeague struct {
	Website     string
	Name        string
	Address     string
	Email       string
	PhoneNumber string
	Fax         string
}

func newLCAPBLeague() (leagues.League, error) {
	return &lcapbLeague{
		Name:        "Ligue de Pelote Basque de Côte d’Argent",
		Website:     "http://www.lcapb.net/",
		Address:     "Maison Départementale des Sports\n153, rue David Johnston\n33000 Bordeaux",
		Email:       "contact@lcapb.net",
		PhoneNumber: "05 56 00 99 15",
		Fax:         "05 56 00 99 15",
	}, nil
}

func (l *lcapbLeague) Details() map[string]string {
	return map[string]string{
		"Name":        l.Name,
		"Website":     l.Website,
		"Address":     l.Address,
		"Email":       l.Email,
		"PhoneNumber": l.PhoneNumber,
		"Fax":         l.PhoneNumber,
	}
}

func (l *lcapbLeague) Challenges() map[string]string {
	return challenges
}

func (l *lcapbLeague) Levels() map[string]string {
	return levels
}

func (l *lcapbLeague) Disciplines() map[string]string {
	return disciplines
}

func (l *lcapbLeague) Display(challengeID string, disciplineID string, levelID string) error {

	logrus.Debugf("[lcapb] Search results for %s %s %s", challengeID, disciplineID, levelID)
	return leagues.Display(uri, challengeID, disciplineID, levelID)
}
